### PR TITLE
test: added basic example

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

adds a basic example 

### Release required?

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: examples/basic/main.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+resource "ibm_resource_group" "resource_group" {
+  count    = var.resource_group != null ? 0 : 1
+  name     = "${var.prefix}-rg"
+  quota_id = null
+}
+
+data "ibm_resource_group" "existing_resource_group" {
+  count = var.resource_group != null ? 1 : 0
+  name  = var.resource_group
+}
```
Comment (ocofaigh) [inline examples/basic/main.tf:15]: Can we use our resource group module instead?

File: examples/basic/main.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# Provision VPC
+#############################################################################
+
+module "slz_vpc" {
+  source            = "../../"
+  resource_group_id = var.resource_group != null ? data.ibm_resource_group.existing_resource_group[0].id : ibm_resource_group.resource_group[0].id
+  region            = var.region
+  name              = var.name
+  prefix            = var.prefix
+  subnets           = var.subnets
```
Comment (ocofaigh) [inline examples/basic/main.tf:27]: add subnets config directly in here, not as a variable.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

**Comment by rajatagarwal-ibm** [comment]
Ran successful test locally 
image

File: examples/basic/main.tf (hunk 1 above)
Comment (rajatagarwal-ibm) [reply examples/basic/main.tf:11]: fixed for other examples too.

File: examples/basic/variables.tf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  default     = null
+}
+
+variable "resource_tags" {
+  description = "List of Tags for the resource created"
+  type        = list(string)
+  default     = null
+}
+
+variable "access_tags" {
```
Comment (ocofaigh) [inline examples/basic/variables.tf:37]: No need to expose access tags in a basic example

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**